tests/python/unittest/test_numpy_ndarray.py

```diff
@@ -1327,3 +1327,26 @@ def test_np_ndarray_pickle():
             a_load = pickle.load(f)
         same(a.asnumpy(), a_load.asnumpy())
 
+@pytest.mark.parametrize('dtype', [np.float32, np.int32])
+@pytest.mark.parametrize('size', [
+    (3, 4, 5, 6),
+    (2, 10),
+    (15,),
+    ()
+])
+@use_np
+def test_dlpack(dtype, size):
+    a = mx.np.random.uniform(size=size)
+    a_np = a.copy()
+
+    pack = mx.npx.to_dlpack_for_read(a)
+    b = mx.npx.from_dlpack(pack)
+
+    a_copy = a.copy()
+    pack2 = mx.npx.to_dlpack_for_write(a_copy)
+    c = mx.npx.from_dlpack(pack2)
```

Comment:
Follow the comment above:

I suggest adding the test of `c += 1` and check whether `a_copy` is mutated (test for wait to read).
```python
c = mx.npx.from_dlpack(pack2)
c += 1
same(a_np + 1, b)
same(a_np + 2, a_copy)
```

Reply:
To put it together:
```python
a = mx.np.random.uniform(size=size)
a_np = a.copy()
a += 1
pack = mx.npx.to_dlpack_for_read(a)
b = mx.npx.from_dlpack(pack)

a_copy = a.copy()
pack2 = mx.npx.to_dlpack_for_write(a_copy)
c = mx.npx.from_dlpack(pack2)
c += 1

del a, pack, pack2
same(a_np + 1, b)
same(a_np + 2, c)
same(a_np + 2, a_copy)
```